content/automate/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -224,39 +224,39 @@ def provision_type(resource)
   end
 end
 
-def cloud_storage(flavor, dialog_array, resource)
-  return unless flavor
-
-  storage = if provision_type(resource) == 'google'
-              get_option_value(resource, :boot_disk_size).gigabytes
+def cloud_storage(args_hash)
+  return unless args_hash[:flavor]
+  storage = if provision_type(args_hash[:resource]) == 'google'
+              get_option_value(args_hash[:resource], :boot_disk_size).gigabytes
             else
-              flavor.root_disk_size.to_i + flavor.ephemeral_disk_size.to_i + flavor.swap_disk_size.to_i
+              args_hash[:flavor].root_disk_size.to_i + args_hash[:flavor].ephemeral_disk_size.to_i + args_hash[:flavor].swap_disk_size.to_i
             end
-  default_option(storage, dialog_array)
+  $evm.log(:info, "Retrieving cloud storage #{storage}")
+  default_option((storage * args_hash[:number_of_vms]), args_hash[:options_array])
 end
 
-def cloud_number_of_cpus(flavor, dialog_array)
-  return unless flavor
-  $evm.log(:info, "Retrieving cloud flavor #{flavor.name} cpus => #{flavor.cpus}")
-  default_option(flavor.cpus, dialog_array)
+def cloud_number_of_cpus(args_hash)
+  return unless args_hash[:flavor]
+  $evm.log(:info, "Retrieving cloud flavor #{args_hash[:flavor].name} cpus => #{args_hash[:flavor].cpus}")
+  default_option((args_hash[:flavor].cpus * args_hash[:number_of_vms]), args_hash[:options_array])
```

Comment:
Two comments that apply to the 3 cloud calculation methods update in this PR.

1. The methods `cloud_number_of_cpus`, `cloud_vm_memory` and `cloud_storage` all start with the condition `return unless args_hash[:flavor]`.  These methods are only called from one method `cloud_value`.

I would move this check into the caller and the methods can assume that they are being called only when a flavor is present in the hash.

2. Constantly dereferencing `args_hash[:flavor]` is not efficient and makes the logic harder to read.  Change the methods to set the variable `flavor` at the top of the methods.

As an example, with these two changes the `cloud_vm_memory` method would become:
```ruby
def cloud_vm_memory(args_hash)
  flavor = args_hash[:flavor]

  $evm.log(:info, "Retrieving flavor #{flavor.name} memory => #{flavor.memory}")
  default_option((flavor.memory * args_hash[:number_of_vms]), args_hash[:options_array])
end
```
Note: Since the other two `args_hash` values are only referenced once I did not assign them to variables.